force-app/main/default/classes/EditDistanceCalculator.cls

```diff
@@ -0,0 +1,45 @@
+public with sharing class EditDistanceCalculator 
+{
+    @TestVisible
+    private static Integer calculateEditDistance( String str1, String str2, Integer m, Integer n ) //NOPMD -  Have to use long parameter here
+    {
+		Contract.requires( ( str1 != null ), 'calculateEditDistance called with null string 1' );
+
+        Contract.requires( ( str2 != null ), 'calculateEditDistance called with null string 2' );
+
+        Contract.requires( ( m != null ), 'calculateEditDistance called with null length of string 1' );
+
+        Contract.requires( ( n!= null ), 'calculateEditDistance called with null length of string 2' );
+
+        if ( m == 0 )  // if str1 is empty - we will have to insert all values to str2 i.e insert operations = str2.length()
+        {
+           return n; 
+        }
+
+        if ( n == 0 ) // if str2 is empty - we will have to delete all values from str2 i.e delete operations = str1.length()
+        {
+            return m;
+        }
+
+        if ( str1.charAt( m - 1 ) == str2.charAt( n - 1 ) ) //if characters match 
+        {
+           return calculateEditDistance( str1, str2, m - 1, n - 1 );  // no change to edit distance: edit distance = previous
+        }
+
```

Comment:
what if the characters dont match?

Reply:
Then it will move on to the next condition

Reply:
It will not execute the if-statement within this if-condition and move on to the next statement  which is to return minimum of the values from the specified indexes